docs/model_developers/handlers.rst

```diff
@@ -95,6 +95,62 @@ Each Handler should override 4 methods of the CRUDHandler:
 The context (See :class:`~inmanta.agent.handler.HandlerContext`) passed to most methods is used to
 report results, changes and logs to the handler and the server.
 
+
+.. _Using facts:
+
+Using facts
+"""""""""""
+
+Facts are properties of the environment whose values are not managed by the orchestrator.
+Facts are either used as input in a model, e.g. a virtual machine provider provides an ip and the model then uses this
+ip to run a service, or used for reporting purposes.
+
+Retrieving a fact in the model is done with the `std::getfact() <../../../reference/modules/std.html#std.getfact>`_
+function.
+
+Example taken from the `openstack Inmanta module <https://github.com/inmanta/openstack>`_:
+
+.. code-block:: inmanta
+    :linenos:
+
+    implementation fipAddr for FloatingIP:
+        self.address = std::getfact(self, "ip_address")
+    end
+
+Setting a value for a fact is done in the handler with the :meth:`~inmanta.agent.handler.HandlerContext.set_fact`
+method. e.g.:
+
+.. code-block:: inmanta
+    :linenos:
+
+    @provider("openstack::FloatingIP", name="openstack")
+    class FloatingIPHandler(OpenStackHandler):
+        def read_resource(self, ctx: handler.HandlerContext, resource: FloatingIP) -> None:
+            ...
+
+        def create_resource(self, ctx: handler.HandlerContext, resource: FloatingIP) -> None:
+            ...
+            # Setting fact manually
+            for key, value in ...:
+                ctx.set_fact(fact_id=key, value=value, expires=True)
+
+By default, facts expire when they are not being refreshed or updated for a certain time, controlled by the
+:inmanta.config:option:`server.fact-expire` config option. Querying for an expired fact will force the
+agent to refresh it first.
+Expired facts are also refreshed periodically. This time interval is controlled by the
+:inmanta.config:option:`server.fact-renew` config option.
+
+
+When reporting a fact, setting the ``expires`` parameter to ``False`` will ensure that this fact never expires. This
```

Comment:
I would also explain the `get_fact` method and explain there are two modes (push and pull) and that if the fact expires, it must also be returned by `get_fact`